lib/MusicBrainz/Server/Report/CatNoLooksLikeLabelCode.pm

```diff
@@ -0,0 +1,35 @@
+package MusicBrainz::Server::Report::CatNoLooksLikeLabelCode;
+use Moose;
+
+with 'MusicBrainz::Server::Report::ReleaseReport',
+     'MusicBrainz::Server::Report::FilterForEditor::ReleaseID';
+
+sub component_name { 'CatNoLooksLikeLabelCode' }
+
+sub query {
+    "
+        SELECT
+            r.id AS release_id, rl.catalog_number,
+            row_number() OVER (ORDER BY ac.name COLLATE musicbrainz, r.name COLLATE musicbrainz)
+        FROM
+            release_label rl
+            JOIN release r
+            ON r.id = rl.release
+            JOIN artist_credit ac ON r.artist_credit = ac.id
+        WHERE rl.catalog_number ~ '^LC[\s-]*[0-9]{4,5}\$'
```

Comment:
Oops, the `\` probably needs to be escaped in Perl double quotes `"…"` for space-matching class escape `\s`.
By the way, the same notation can be used to match digits:
```suggestion
        WHERE rl.catalog_number ~ '^LC[\\s-]*\\d{4,5}\$'
```